src/code/z_parameter.c

```diff
@@ -330,9 +330,9 @@ u8 Item_Give(PlayState* play, u8 item) {
         return ITEM_NONE;
 
     } else if ((item == ITEM_HEART_PIECE_2) || (item == ITEM_HEART_PIECE)) {
-        gSaveContext.save.inventory.questItems += (1 << QUEST_HEART_PIECE_COUNT);
-        if ((GET_SAVE_INVENTORY_QUEST_ITEMS & 0xF0000000) == (4 << QUEST_HEART_PIECE_COUNT)) {
-            gSaveContext.save.inventory.questItems ^= (4 << QUEST_HEART_PIECE_COUNT);
+        INCREMENT_QUEST_HEART_PIECE_COUNT;
+        if (CHECK_EQ_QUEST_HEART_PIECE_COUNT(4)) {
+            RESET_HEART_PIECE_COUNT(4);
```

Comment:
This usage looks a bit strange, is there any reason you'd reset to not-4?

Reply:
I was thinking it was to parallel `CHECK_EQ_QUEST_HEART_PIECE_COUNT` and also could be something like TP where it's 5 HP's per Heart Upgrade, but I suppose there'd need to be a lot more changes for something like that. I can just keep it to `RESET_HEART_PIECE_COUNT` and hide that detail away.

`CHECK_EQ_QUEST_HEART_PIECE_COUNT(4)` -> `CHECK_MAX_QUEST_HEART_PIECE_COUNT`
`RESET_HEART_PIECE_COUNT(4)` -> `RESET_HEART_PIECE_COUNT`
Perhaps?

Reply:
Yeah, you're not going to be able to edit the numbers without rewriting the whole system, basically.